### ISSUE TITLE
[Feature request] Show pin icon only on hover to save some space

### ISSUE THREAD

**Issue by pawelzwronek** [notepad-plus-plus/notepad-plus-plus#15815]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Description of the Issue

The Pin tab feature is great, and I know I'll be using it often. The issue is it comes with taking precious space on tabbar: 

![obraz](https://github.com/user-attachments/assets/5b96c8f7-045d-4e25-9882-5c8db74a63b3)

### Describe the solution you'd like.

I propose to show the pin icon on unpinned and current tabs, only when hovering mouse like so:

![obraz](https://github.com/user-attachments/assets/0730d5cb-ce5b-487f-9798-263226ef54ec)
When I hove mouse over a tab I dont care the name, so the pin icon could overlay the end of the name label. This way tabs length is preserved. Further, an overlayed pin icon could have few pixels wide horizontal gradient for more pleasing appearance:

![obraz](https://github.com/user-attachments/assets/2eebf00f-985b-4c73-b4c7-c87c0e73eda1)

If you like the idea, I could code it myself if you will.
 

### Debug Information

```shell
Not relevant.
```

**Comment by donho** [comment]
@pawelzwronek 
Not clear to me.

> I propose to show the pin icon on unpinned and current tabs, only when hovering mouse like so:

It's the case in the master, isn't it?

**Comment by donho** [comment]
> When I hove mouse over a tab I dont care the name, so the pin icon could overlay the end of the name label. This way tabs length is preserved. Further, an overlayed pin icon could have few pixels wide horizontal gradient for more pleasing appearance:

I don't like the idea. Let's see the opinion of user about current pin feature in the next release.

**Comment by alankilborn** [comment]
I don't like this idea either.  If I'm understanding correctly, it would cause tabs to change in horizontal size as the mouse moves over them.  In multiline mode this could even cause tabs to jump lines as the mouse pans across the tabs.  It sounds like a "disconcerting" user experience, but I suppose if someone wants to demo it for others to evaluate, then we'd certainly do that and provide feedback.

**Comment by pawelzwronek** [comment]
@alankilborn no no. I must have put my proposition unclear way. Hovering mouse doesn't change the size of the tabs, it will only make the pin icon to appear and overlay the end of the label, like on the zoomed picture.

**Comment by alankilborn** [comment]
>no no..

OK, sorry, I didn't read far enough into your suggestion.  (embarrassed)

You are right in that tabs really do appear too wide, because space is left for the pin and the close button.
Your suggestion saves that space, but, without seeing it in action I'm unsure if I would like the effect or not.
Again, I encourage you to provide a demo for it, if you can spare the time (and you don't mind that time being wasted if it isn't accepted into Notepad++).

**Comment by pawelzwronek** [comment]
This is how I image it might work:
![Animation](https://github.com/user-attachments/assets/6308046c-0efc-4cd1-9201-d2a301080329)

**Comment by pawelzwronek** [comment]
I notices Visual Studio behave like current master branch. The benefit of this might be more pleasing appearance because of extra padding, but it's side effect of reserving extra space for the pin. At first, when I enabled pinning feature I was shocked that an extra space was taken "just" to show the pin and I hesitated if I want to enable it at this cost. I think more users will have the same dilemma. We need to wait for the comments.
